dateparser/search/search.py

```diff
@@ -0,0 +1,189 @@
+                            possible_substrings.append(current_substrings)
+                        parsed_best, substrings_best = self.choose_best_split(possible_parsed, possible_substrings)
+                        for k in range(len(parsed_best)):
+                            if parsed_best[k]['date_obj']:
+                                parsed.append(parsed_best[k])
+                                substrings.append(substrings_best[k])
+        parser._settings.RELATIVE_BASE = None
+        return parsed, substrings
+
+    @apply_settings
+    def search_parse(self, shortname, text, settings=None):
+        translated, original = self.search(shortname, text, settings)
+        if shortname not in ['vi', 'hu']:
+            parser = DateDataParser(languages=['en'], settings=settings)
+            parsed, substrings = self.parse_found_objects(parser=parser, to_parse=translated,
+                                                          original=original, translated=translated, settings=settings)
+        else:
+            parser = DateDataParser(languages=[shortname], settings=settings)
+            parsed, substrings = self.parse_found_objects(parser=parser, to_parse=original,
+                                                          original=original, translated=translated, settings=settings)
+        return list(zip(substrings, [i['date_obj'] for i in parsed]))
+
+
+class DateSearchWithDetection:
+    def __init__(self):
+        self.loader = LanguageDataLoader()
+        self.available_language_map = self.loader.get_language_map()
+        self.search = ExactLanguageSearch(self.loader)
+
+    def detect_language(self, text, languages):
+        if isinstance(languages, (list, tuple, collections.Set)):
+
+            if all([language in self.available_language_map for language in languages]):
+                languages = [self.available_language_map[language] for language in languages]
+            else:
+                unsupported_languages = set(languages) - set(self.available_language_map.keys())
+                raise ValueError(
+                    "Unknown language(s): %s" % ', '.join(map(repr, unsupported_languages)))
+        elif languages is not None:
+            raise TypeError("languages argument must be a list (%r given)" % type(languages))
+
+        if languages:
+            self.language_detector = FullTextLanguageDetector(languages=languages)
+        else:
+            self.language_detector = FullTextLanguageDetector(list(self.available_language_map.values()))
+
+        return self.language_detector._best_language(text)
+
+    @apply_settings
+    def search_dates(self, text, languages=None, settings=None):
+        language_shortname = self.detect_language(text=text, languages=languages)
+        if not language_shortname:
+            return {'Language': None, 'Dates': None}
+        return {'Language': language_shortname, 'Dates': self.search.search_parse(language_shortname, text, settings)}
```

Comment:
This should read: `self.search.search_parse(language_shortname, text, settings=settings)`, otherwise the `@apply_settings` wrapper will crash.